internal/state/v1/state.go

```diff
@@ -0,0 +1,260 @@
+package v1
+
+import (
+	"fmt"
+	"log/slog"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/freeverseio/laos-universal-node/internal/platform/storage"
+	"github.com/freeverseio/laos-universal-node/internal/scan"
+	"github.com/freeverseio/laos-universal-node/internal/state"
+	"github.com/freeverseio/laos-universal-node/internal/state/enumerated"
+	"github.com/freeverseio/laos-universal-node/internal/state/enumerated_total"
+	"github.com/freeverseio/laos-universal-node/internal/state/ownership"
+)
+
+type stateService struct {
+	storageService storage.Service
+}
+
+func NewStateService(storageService storage.Service) state.StateService {
+	return &stateService{
+		storageService: storageService,
+	}
+}
+func (s *stateService) NewTransaction() state.Tx {
+
+	storageTx := s.storageService.NewTransaction()
+	return &tx{
+		ownershipTrees:       make(map[common.Address]ownership.Tree),
+		enumeratedTrees:      make(map[common.Address]enumerated.Tree),
+		enumeratedTotalTrees: make(map[common.Address]enumerated_total.Tree),
+		Tx:                   storageTx}
+}
+
+type tx struct {
+	Tx                   storage.Tx
```

Comment:
Make this unexported, please. If you want to avoid `tx.tx` you can also rename it

Reply:
changed

Reply:
I changed it to unexported. If you don't insist on naming I would keep it as tx. I like to keep this tx on all levels.